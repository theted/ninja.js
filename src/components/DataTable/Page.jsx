import React from 'react';
import PropTypes from 'prop-types';

const Page = ({ pageNumber, currentPageNumber, onChange }) => {
  const isActivePage = () => currentPageNumber === pageNumber;

  const renderedPageNumber = () => pageNumber + 1;

  const click = (event) => {
    event.preventDefault();
    onChange(pageNumber);
  };

  const className = `page-link${isActivePage()}` ? ' button-outline' : '';

  return (
    <li className="page-item mr-1">
      <button className={className} onClick={click} type="button">
        {renderedPageNumber()}
      </button>
    </li>
  );
};

Page.propTypes = {
  pageNumber: PropTypes.number,
  currentPageNumber: PropTypes.number,
  onChange: PropTypes.func,
};

Page.defaultProps = {
  pageNumber: 0,
  currentPageNumber: 0,
  onChange: () => {},
};

export default Page;
