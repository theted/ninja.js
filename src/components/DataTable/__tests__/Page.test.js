import React from 'react';
import { shallow } from 'enzyme';
import Page from '../Page';

describe('Page component', () => {
  it('renders without crashing', () => {
    shallow(<Page />);
  });

  it('matches snapshot', () => {
    const component = shallow(<Page />);
    expect(component).toMatchSnapshot();
  });
});
