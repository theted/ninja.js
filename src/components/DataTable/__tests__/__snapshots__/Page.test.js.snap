// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page component matches snapshot 1`] = `ShallowWrapper {}`;
